Lib/socket.py

```diff
@@ -356,7 +356,7 @@ def _sendfile_use_sendfile(self, file, offset=0, count=None):
                 raise _GiveupOnSendfile(err)  # not a regular file
             if not fsize:
                 return 0  # empty file
-            blocksize = fsize if not count else count
+            blocksize = min(count or fsize, 2 ** 30)  # max 1GiB
```

Comment:
Do we have to truncate the size on 64-bit system? Please add a comment. I suggest to write this instruction in two limits, to make the purpose of each line more explicit. Lilke:

```
blocksize = fsize if not count else count
# bpo-38319: truncate to 1 GiB to prevent OverflowError
blocksize = min(blocksize, 2 ** 30)
```